core/GlobalState.h

```diff
@@ -532,6 +532,25 @@ class GlobalState final {
         return this->files->span();
     }
 
+    const UnorderedSet<core::SymbolRef> &getSymbolsReferencedByFile(core::FileRef fref) const {
+        ENFORCE(packageDB().genPackages());
+        ENFORCE(symbolsReferencedByFile.size() == this->files->size(),
+                "mismatch in files.size ({}) and symbolsReferencedByFile.size(): ({})", files->size(),
+                symbolsReferencedByFile.size());
+        auto ix = fref.id();
+        ENFORCE(ix < symbolsReferencedByFile.size());
+        return this->symbolsReferencedByFile[ix];
+    }
+
+    void setSymbolsReferencedByFile(core::FileRef fref, UnorderedSet<core::SymbolRef> &referencedSymbols) {
```

Comment:
Should this take an rvalue-reference?

I _think_ that the swap call on 551 is going to copy the `referencedSymbols`